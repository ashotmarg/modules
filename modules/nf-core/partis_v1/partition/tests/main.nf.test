// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test partis/partition
nextflow_process {

    name "Test Process PARTIS_PARTITION"
    script "../main.nf"
    process "PARTIS_PARTITION"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "partis"
    tag "partis/partition"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("partis_partition") {

        // TODO nf-core: If you are created a test for a chained module
        // (the module requires running more than one process to generate the required output)
        // add the 'setup' method here.
        // You can find more information about how to use a 'setup' method in the docs (https://nf-co.re/docs/contributing/modules#steps-for-creating-nf-test-for-chained-modules).

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/partis/partisTestSmall.fa", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.version).match("version") },
                //{ assert process.out.yaml.get(0).get(3) ==~ ".*/example.vcf.pvar" }
                { assert file(process.out.yaml.get(1).get(0).find { file(it).name == "test.yaml" }).exists() }
                //{ assert file(process.out.yaml.get(0).get(1).find { file(it).name == "test.yaml" }).exists() }
                //{ assert workflow.trace.tasks().size() == 3 }
                //{ assert process.trace.tasks().size() == 3 }
                //{ assert file(process.out.yaml.get(0).get(1).find { file(it).name == "test.yaml" }).exists() }
                //{ assert file(process.out.yaml[1].find { file(it).name == "test.yaml" }).exists() }
                //{ assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

}

