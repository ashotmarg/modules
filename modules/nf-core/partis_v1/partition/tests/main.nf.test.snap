{
    "version": {
        "content": null,
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-10T23:41:45.979442"
    }
}